website/content/projects/2019/spec.md

```diff
@@ -0,0 +1,386 @@
+*   **UI**: Besides the command-line interface, games may have additional UI elements, like visualizing a map of the game or displaying graphics associated with individual rooms. This component would be responsible for any non-CLI elements of the user interface.
+
+Each team will be assigned to work on one component (teams will have a chance to express a preference for certain components, although we can’t guarantee we’ll be able to assign each team to their preferred component). In each team, a TA will act as Senior Developer.
+
+The project will span seven weeks (Weeks 4 - 10), with a final review and team presentations taking place during Finals Week. The seven weeks of the project will be divided into a project kick-off and four sprints, with sprint reviews happening on Mondays.
+
+
+
+*   Week 4: Project kick-off. This week will be primarily used for you to familiarize yourself with project and the exact work your team will do.
+*   Week 5: Sprint 1
+*   Week 6: Sprint 2. A basic prototype should ideally be done by the end of this sprint.
+*   Weeks 7-8: Sprint 3
+*   Weeks 9-10: Sprint 4
+
+
+<!--
+## Suggested initial goals
+
+TODO
+-->
+
+# Project Evaluation
+
+The project is worth 70% of your final grade, and these 70 points are divided as follows:
+
+*   5: Team Design Exercise
+*   40: Your individual score on the project
+*   15: Your team score (the average of all the individual scores on a team)
+*   10: Final presentation during Finals Week.
+
+Unlike a traditional programming assignment or project, the exact final outcome of the project is not known in advance. Of course, we do have high-level goals and requirements (which have been described above) and, while those requirements will help us measure whether the project is “done”, they are not a strict rubric. As we saw in class, this is one of the strengths (and challenges) of agile software development: you can re-evaluate requirements and make design changes quickly, but there is no detailed requirements document that you can use to check whether you’ve contractually fulfilled your obligations to a customer.
+
+As such, your individual score on the project will be based on how well you perform certain tasks that you will do throughout the project (and which you would encounter in any real software project). More specifically, you will be evaluated based on the following:
+
+
+
+*   How well you document tasks/issues (and whether you complete the tasks you commit to doing in a given sprint).
+*   Whether you get pull requests to master approved
+*   How well you do code reviews (on pull requests where you are a reviewer) 
+*   How well you do your peer evaluations
+
+Note: no part of your score will be determined by your teammates. So, when you receive a code review or a peer evaluation, we will be evaluating the quality of that code review / peer evaluation. We will not be using that review/evaluation to determine your score.
```

Comment:
I think the subject of the review here got a little lost. I think we need to explicitly say that we will be evaluating the person doing the reviewing.

Reply:
I've changed this to:

> Note: no part of your score will be determined by your teammates. So, when you receive a code review or a peer evaluation, we will be evaluating the quality of the code review / peer evaluation itself; we will not be using that review/evaluation to determine your score. Please note that, in a peer evaluation, we will specifically be evaluating the person giving the feedback (by looking at the quality of the feedback they provide).

Code reviews are a little different, because they're evaluated like pull requests, and the person receiving the code review also puts in some effort (to apply the requested changes, etc.), which is why I specified that they get points for the code review as well.